src/corehaptics.cs

```diff
@@ -0,0 +1,398 @@
+
+		[Export ("isMutedForHaptics")]
+		bool IsMutedForHaptics { get; set; }
+
+		[Export ("autoShutdownEnabled")]
+		bool AutoShutdownEnabled { [Bind ("isAutoShutdownEnabled")] get; set; }
+
+		[Export ("initAndReturnError:")]
+		[DesignatedInitializer]
+		IntPtr Constructor ([NullAllowed] out NSError error);
+
+		[NoMac]
+		[Export ("initWithAudioSession:error:")]
+		[DesignatedInitializer]
+		IntPtr Constructor ([NullAllowed] AVAudioSession audioSession, [NullAllowed] out NSError error);
+
+		[Async]
+		[Export ("startWithCompletionHandler:")]
+		void Start ([NullAllowed] Action<NSError> completionHandler);
+
+		[Export ("startAndReturnError:")]
+		bool Start ([NullAllowed] out NSError outError);
+
+		[Async]
+		[Export ("stopWithCompletionHandler:")]
+		void Stop ([NullAllowed] Action<NSError> completionHandler);
+
+		[Async]
+		[Export ("notifyWhenPlayersFinished:")]
+		void NotifyWhenPlayersFinished (Action<NSError> finishedHandler);
+
+		[Export ("createPlayerWithPattern:error:")]
+		[return: NullAllowed]
+		ICHHapticPatternPlayer CreatePlayer (CHHapticPattern pattern, [NullAllowed] out NSError outError);
+
+		[Export ("createAdvancedPlayerWithPattern:error:")]
+		[return: NullAllowed]
+		ICHHapticAdvancedPatternPlayer CreateAdvancedPlayer (CHHapticPattern pattern, [NullAllowed] out NSError outError);
+
+		[Export ("registerAudioResource:options:error:")]
+		nuint RegisterAudioResource (NSUrl resourceUrl, NSDictionary options, [NullAllowed] out NSError outError);
+
+		[Export ("unregisterAudioResource:error:")]
+		bool UnregisterAudioResource (nuint resourceId, [NullAllowed] out NSError outError);
+
+		[Export ("playPatternFromURL:error:")]
+		bool PlayPattern (NSUrl fileUrl, [NullAllowed] out NSError outError);
+
+		[Export ("playPatternFromData:error:")]
+		bool PlayPattern (NSData data, [NullAllowed] out NSError outError);
+	}
+
+	[Static]
+	[Internal]
+	[Mac (10,15), iOS (13,0)]
+	partial interface CHHapticPatternDefinitionKeys {
+		[Field ("CHHapticPatternKeyVersion")]
+		NSString VersionKey { get; }
+
+		[Field ("CHHapticPatternKeyPattern")]
+		NSString PatternKey { get; }
+
+		[Field ("CHHapticPatternKeyEvent")]
+		NSString EventKey { get; }
+
+		[Field ("CHHapticPatternKeyEventType")]
+		NSString EventTypeKey { get; }
+
+		[Field ("CHHapticPatternKeyTime")]
+		NSString TimeKey { get; }
+
+		[Field ("CHHapticPatternKeyEventDuration")]
+		NSString EventDurationKey { get; }
+
+		[Field ("CHHapticPatternKeyEventWaveformPath")]
+		NSString EventWaveformPathKey { get; }
+
+		[Field ("CHHapticPatternKeyEventParameters")]
+		NSString EventParametersKey { get; }
+
+		[Field ("CHHapticPatternKeyParameter")]
+		NSString ParameterKey { get; }
+
+		[Field ("CHHapticPatternKeyParameterID")]
+		NSString ParameterIdKey { get; }
+
+		[Field ("CHHapticPatternKeyParameterValue")]
+		NSString ParameterValueKey { get; }
+
+		[Field ("CHHapticPatternKeyParameterCurve")]
+		NSString WeakCHHapticParameterCurveKey { get; }
```

Comment:
I think this could be just `WeakParameterCurveKey`.

Reply:
This is an internal class used as the holder of the key for the strong dictionary. This means that I'd have to add an extra [Export ("TheNewKeyName")] in  CHHapticPatternDefinition. For that matter, we might as well remove the Weak, call it ParameterCurveKey and leave it cleaner.

Reply:
not sure I understand

the problem is having the prefix/part of the type name after the `Weak`

Reply:
That there is a better, way, let me push the diff, better with code